cli/ballerina-cli-utils/src/main/ballerina/packaging_pull/packaging_pull.bal

```diff
@@ -50,23 +67,33 @@ function pullPackage (http:Client definedEndpoint, string url, string dirPath, s
     match result {
         http:Response response => httpResponse = response;
         error e => {
-            io:println(logger.formatLog("connection to the remote host failed : " + e.message));
-            return;
+            io:println(logFormatter.formatLog("connection to the remote host failed : " + e.message));
+            return 1;
```

Comment:
Shall we create an error here, and return it without printing? Who ever calling this function can decide how to handle it. In our case, caller should log it. In that way, we are logging only at one place.

Also then the function signature should be `function pullPackage(...) returns error?`. That is much cleaner IMO than returning a flag.